### PR TITLE
Add routes map

### LINKED ISSUE #18: Move routes to constants
Create an object that contains app routes and use it in application

### PR DESCRIPTION
**Pull request by ekostandov**

### CODE REVIEW

File: src/constants/routes.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const routes = {
```
Comment (singvarr) [inline src/constants/routes.js]: should be default export

File: src/constants/routes.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  home: '/',
```
Comment (singvarr) [inline src/constants/routes.js]: all object field names should be in SCREAMING_SNAKE_CASE

File: src/constants/routes.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  cart: '/cart',
+  products: '/products',
```
Comment (singvarr) [inline src/constants/routes.js]: all pathnames should be in kebab-case

File: src/App/Header/Logo.js [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
 import logo from './logo.png';
+import { routes } from 'constants/routes';
 
 function Logo() {
   return (
     <div className='logo'>
-      <Link to='/'>
-        <img src={logo} alt='' />
+      <Link to={routes.home}>
+        <img src={logo} alt='Logo' />
```
Comment (singvarr) [inline src/App/Header/Logo.js]: remove redundant diff

**Review by singvarr — CHANGES_REQUESTED** [review]
Also, please, check other files that use routes (for example, `src/App/Main/Cart/CartPage.js`, `src/App/Main/Main.js` or `src/App/Main/Product/ProductListItem.js`)

File: src/App/Header/Logo.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
 import { Link } from 'react-router-dom';
 import logo from './logo.png';
+import { routes } from 'constants/routes';
```
Comment (singvarr) [inline src/App/Header/Logo.js]: Absolute import should be before a relative one

File: src/constants/routes.js (hunk 1 above)
Comment (ekostandov) [reply src/constants/routes.js]: Done.

File: src/constants/routes.js (hunk 2 above)
Comment (ekostandov) [reply src/constants/routes.js]: Done.

File: src/constants/routes.js (hunk 3 above)
Comment (ekostandov) [reply src/constants/routes.js]: Done.

File: src/App/Header/Logo.js (hunk 4 above)
Comment (ekostandov) [reply src/App/Header/Logo.js]: Done.

File: src/App/Header/Logo.js (hunk 5 above)
Comment (ekostandov) [reply src/App/Header/Logo.js]: Done.

File: src/App/Main/Product/ProductListItem.js [hunk 6]
```diff
@@ -91,7 +92,7 @@ class ProductListItem extends Component {
           {isLiked ? <span> &#9829; </span> : <span> &#9825; </span>}
         </button>
         <h2 className='product-title'>
-          <Link to={`/products/${id}`}>{name}</Link>
+          <Link to={`${routes.PRODUCTS}/${id}`}>{name}</Link>
```
Comment (singvarr) [inline src/App/Main/Product/ProductListItem.js]: Should be a separate pathname - `/products/:id`. Use `generatePath` for construction of url from pattern

**Review by singvarr — CHANGES_REQUESTED** [review]
Fix app build

File: src/constants/routes.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const routes = {
+  HOME: '/',
+  CART: '/cart',
+  PRODUCTS: '/products/:id',
```
Comment (singvarr) [inline src/constants/routes.js]: In the app there were a page with products and a page with a separate product. I think we need to keep two separate constants here

File: src/App/Main/Product/ProductListItem.js [hunk 8]
```diff
@@ -2,6 +2,8 @@ import { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Link } from 'react-router-dom';
+import { generatePath } from 'react-router';
```
Comment (singvarr) [inline src/App/Main/Product/ProductListItem.js]: Should be imported from `react-router-dom`. `react-router-dom` is default package that reexports all stuff from `react-router` and contains DOM bindings

File: src/App/Main/Product/ProductListItem.js [hunk 9]
```diff
@@ -91,7 +93,7 @@ class ProductListItem extends Component {
           {isLiked ? <span> &#9829; </span> : <span> &#9825; </span>}
         </button>
         <h2 className='product-title'>
-          <Link to={`/products/${id}`}>{name}</Link>
+          <Link to={generatePath(routes.PRODUCTS, { id: id })}>{name}</Link>
```
Comment (singvarr) [inline src/App/Main/Product/ProductListItem.js]: Use shorthand form for object with params

File: src/App/Header/Mainmenu/Mainmenu.js [hunk 10]
```diff
@@ -7,16 +8,16 @@ function Mainmenu() {
     <div className='menu'>
       <ul>
         <li>
-          <Link to='/'>Home</Link>
+          <Link to={routes.HOME}>Home</Link>
         </li>
         <li>
-          <Link to='/products'>Product</Link>
+          <Link to={routes.PRODUCTS}>Product</Link>
```
Comment (singvarr) [inline src/App/Header/Mainmenu/Mainmenu.js]: Here was another page. Before your change pattern was `/products`, now it's `/products/:id`

**Review by singvarr — COMMENTED** [review]
Check `src/App/Main/Main.js`, please. It contains magic strings that we need to replace by constants

File: src/App/Main/Product/ProductListItem.js (hunk 9 above)
Comment (singvarr) [reply src/App/Main/Product/ProductListItem.js]: Also find and add ESLint rule that will validate this

**Review by singvarr — APPROVED** [review]

**Pull request merged**